server/src/server.ts

```diff
@@ -1013,7 +1052,10 @@ function onMessage(msg: p.Message) {
       // TODO: close watcher when lang-server shuts down. However, by Node's
       // default, these subprocesses are automatically killed when this
       // language-server process exits
-      let found = utils.findNodeBuildOfProjectRoot(projectRootPath);
+      let found = utils.findNodeBuildOfProjectRoot(
+        projectRootPath,
+        extensionConfiguration.binaryPath
```

Comment:
There are all these extra parameters added. Are they necessary?
Can `findNodeBuildOfProjectRoot` use `extensionConfiguration.binaryPath` directly without passing it as an argument?

Reply:
The function `findNodeBuildOfProjectRoot` is defined in the `server/src/utils.ts` and I am not sure how can it uses the `extensionConfiguration.binaryPath` without passing as an argument because the configuration is defined here in the `server/src/server.ts` :(

Reply:
I can add a new function something like `findBinaryFromConfig` and use it. It will allow to not use of an extra argument in the `findNodeBuildOfProjectRoot`.

I will try

Reply:
done